results/pkg/api/server/v1alpha1/server.go

```diff
@@ -401,6 +419,18 @@ func (s Server) getResultByID(name string) (*pb.Result, error) {
 	return result, nil
 }
 
+func getResultETag(result *pb.Result) (etag string, err error) {
```

Comment:
Avoid naming parameters unless it helps identify what's being returned (e.g. if you're returning multiple things of the same type).

This can often make things harder to read and mask errors. In this case, it's actually making it harder to see that this func will never return an err since the value is scoped to the if statement (https://play.golang.org/p/MVnMta2Heme for a simplified example).